plugins/zify/zify.ml

```diff
@@ -0,0 +1,1158 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Constr
+open Names
+open Pp
+open Lazy
+
+(* From Coqlib *)
+let make_dir l = DirPath.make (List.rev_map Id.of_string l)
+
+(** [get_type_of] performs beta reduction ;
+    Is it ok for Retyping.get_type_of (Zpower_nat n q) to return (fun _ : nat => Z) q ? *)
```

Comment:
Retyping.get_type_of may return types which are not (beta) reduced.
This is probably a feature but I was puzzled because the term was simply typed.
This  was responsible for a bug because Z was expected...